### PR TITLE
Fix render template for number template parts (#9406)

### LINKED ISSUE #9406: Display mode Related Values fails when rendering number
### Preflight Checklist

- [X] I have completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).
- [X] I'm on [the latest version of Directus](https://github.com/directus/directus/releases).
- [X] There's [no other issue](https://github.com/directus/directus/issues) that already describes my problem.

### Describe the Bug

render-template.vue may call translate function with a number. This cause an error in console and editor becomes stale (impossible to close it).

https://github.com/directus/directus/blob/62e5d22e3b76664b1c53e6883ffc93385bfe3e64/app/src/views/private/components/render-template/render-template.vue#L17

### To Reproduce

- Define a field display with Related Values and reference a number field in the template.
- Edit an item from this collection.

### Errors Shown

```
translate-literal.ts:6 Uncaught (in promise) TypeError: literal.startsWith is not a function
    at Proxy.translate (translate-literal.ts:6)
    at render-template.vue:17
    at renderList (runtime-core.esm-bundler.js:5763)
    at Proxy._sfc_render (render-template.vue:3)
    at renderComponentRoot (runtime-core.esm-bundler.js:460)
    at ReactiveEffect.componentUpdateFn [as fn] (runtime-core.esm-bundler.js:4227)
    at ReactiveEffect.run (reactivity.esm-bundler.js:160)
    at setupRenderEffect (runtime-core.esm-bundler.js:4353)
    at mountComponent (runtime-core.esm-bundler.js:4136)
    at processComponent (runtime-core.esm-bundler.js:4094)
```

### What version of Directus are you using?

v9.0.0-rc.100 (main branch)

### What version of Node.js are you using?

v16.13.0

### What database are you using?

Postgres 13

### What browser are you using?

Chrome

### What operating system are you using?

Linux

### How are you deploying Directus?

locally (dev)

### PR DESCRIPTION
**Pull request by Toilal**
Fixes #9406

### CODE REVIEW

File: app/src/views/private/components/render-template/render-template.vue [hunk 1]
```diff
@@ -2,7 +2,7 @@
 	<div ref="templateEl" class="render-template">
 		<span class="vertical-aligner" />
 		<template v-for="(part, index) in parts" :key="index">
-			<value-null v-if="part === null || part.value === null" />
+			<value-null v-if="part === null || (typeof part === 'object' && part.value === null)" />
```
Comment (joselcvarela) [inline app/src/views/private/components/render-template/render-template.vue:5]:
```suggestion
			<value-null v-if="part === null || part?.value === null" />
```

Comment (Toilal) [reply app/src/views/private/components/render-template/render-template.vue:5]: At this point in the template, we are not sure `part` is an object. It seems safer and explicit to keep typeof checking here. It's done on line 8 too.

Comment (joselcvarela) [reply app/src/views/private/components/render-template/render-template.vue:5]: Good point. 
This makes me thinking if we should use more restrictive type checking. As for now, let's keep it that way.

**Comment by Toilal** [comment]
Could you run the ci please ?

**Review by rijkvanzanten — APPROVED** [review]

**Comment by rijkvanzanten** [comment]
@Toilal Even better, I'll run the CI, approve, and merge the PR 👌🏻 🎉

File: app/src/views/private/components/render-template/render-template.vue (hunk 1 above)
Comment (Toilal) [reply app/src/views/private/components/render-template/render-template.vue:5]: [vue-tsc](https://www.npmjs.com/package/vue-tsc) can run tsc typecheck in Vite/Vue3 projects.

**Pull request merged**